lib/streamlit/runtime/secrets.py

```diff
@@ -329,36 +478,36 @@ def __repr__(self) -> str:
         # the file must already exist.
         """A string representation of the contents of the dict. Thread-safe."""
         if not runtime.exists():
-            return f"{self.__class__.__name__}(file_paths={self._file_paths!r})"
-        return repr(self._parse(True))
+            return f"{self.__class__.__name__}"
+        return repr(self._parse(self._print_exceptions_on_access))
```

Comment:
Hm, I wonder if we want to change the argument for `self._parse` to be `suppress_exceptions` instead of `print_exception` since it seems like the much more common case here is to want to print them. I think only the `load_if_toml_exists` actually sets `print_exception` to `False` in the code as it exists today.

Then we'd be able to not pass an arg to `self._parse` and use the default nearly everywhere it gets called, and the `if` statements within the `_parse` method could be `if not suppress_exceptions and self._print_exceptions_on_access` (although I think having both a `suppress_exceptions` argument and `self._print_exceptions_on_access` attribute starts to get a bit confusing. Not sure if there are better ways of naming these things 😬)

Reply:
good idea. i took a stab at this - how's just having one variable - `_supress_print_error_on_exception` that we set to false when running `load_toml_if_exists`